e2e/cephfs.go

```diff
@@ -1962,6 +1962,158 @@ var _ = Describe(cephfsType, func() {
 				}
 			})
 
+			By("checking snapshot creation with backing snapshot as true", func() {
+				totalCount := 2
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					framework.Failf("failed to load PVC: %v", err)
+				}
+				pvc.Namespace = f.UniqueName
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					framework.Failf("failed to create PVC: %v", err)
+				}
+
+				_, pv, err := getPVCAndPV(f.ClientSet, pvc.Name, pvc.Namespace)
+				if err != nil {
+					framework.Failf("failed to get PV object for %s: %v", pvc.Name, err)
+				}
+
+				app, err := loadApp(appPath)
+				if err != nil {
+					framework.Failf("failed to load application: %v", err)
+				}
+				app.Namespace = f.UniqueName
+				app.Spec.Volumes[0].PersistentVolumeClaim.ClaimName = pvc.Name
+				appLabels := map[string]string{
+					appKey: appLabel,
+				}
+				app.Labels = appLabels
+				optApp := metav1.ListOptions{
+					LabelSelector: fmt.Sprintf("%s=%s", appKey, appLabels[appKey]),
+				}
+				err = writeDataInPod(app, &optApp, f)
+				if err != nil {
+					framework.Failf("failed to write data: %v", err)
+				}
+
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.PersistentVolumeClaimName = &pvc.Name
+				for i := 0; i < totalCount; i++ {
+					name := fmt.Sprintf("%s%d", f.UniqueName, i)
+					snap.Name = name
+					err = createSnapshot(&snap, deployTimeout)
+					if err != nil {
+						framework.Failf("failed to create snapshot: %v", err)
+					}
+				}
+				validateCephFSSnapshotCount(f, 1, subvolumegroup, pv)
```

Comment:
`totalCount` is set to 2, and after creating all snapshots you expect only 1. There seem to be 2 snapshots on the CephFS filesystem according to the e2e logs:
```
    [1mSTEP:[0m checking snapshot creation with backing snapshot as true [38;5;243m03/24/23 10:11:11.861[0m
    Mar 24 10:11:11.883: INFO: Waiting up to &PersistentVolumeClaim{ObjectMeta:{csi-cephfs-pvc  cephfs-2540    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []},Spec:PersistentVolumeClaimSpec{AccessModes:[ReadWriteMany],Resources:ResourceRequirements{Limits:ResourceList{},Requests:ResourceList{storage: {{1073741824 0} {<nil>} 1Gi BinarySI},},Claims:[]ResourceClaim{},},VolumeName:,Selector:nil,StorageClassName:*csi-cephfs-sc,VolumeMode:nil,DataSource:nil,DataSourceRef:nil,},Status:PersistentVolumeClaimStatus{Phase:,AccessModes:[],Capacity:ResourceList{},Conditions:[]PersistentVolumeClaimCondition{},AllocatedResources:ResourceList{},ResizeStatus:nil,},} to be in Bound state
    Mar 24 10:11:11.883: INFO: waiting for PVC csi-cephfs-pvc (0 seconds elapsed)
    Mar 24 10:11:13.886: INFO: waiting for PVC csi-cephfs-pvc (2 seconds elapsed)
    Mar 24 10:11:13.890: INFO: Waiting for PV pvc-e4d7c2b7-1e68-40f6-bdea-fa9c86ef3c18 to bind to PVC csi-cephfs-pvc
    Mar 24 10:11:13.890: INFO: Waiting up to timeout=10m0s for PersistentVolumeClaims [csi-cephfs-pvc] to have phase Bound
    Mar 24 10:11:13.891: INFO: PersistentVolumeClaim csi-cephfs-pvc found and phase=Bound (1.428568ms)
    Mar 24 10:11:13.891: INFO: Waiting up to 10m0s for PersistentVolume pvc-e4d7c2b7-1e68-40f6-bdea-fa9c86ef3c18 to have phase Bound
    Mar 24 10:11:13.893: INFO: PersistentVolume pvc-e4d7c2b7-1e68-40f6-bdea-fa9c86ef3c18 found and phase=Bound (1.579854ms)
    Mar 24 10:11:13.903: INFO: Waiting up to csi-cephfs-demo-pod to be in Running state
    Mar 24 10:11:15.909: INFO: ExecWithOptions {Command:[/bin/sh -c dd if=/dev/zero of=/var/lib/www/test bs=1M count=10 status=none] Namespace:cephfs-2540 PodName:csi-cephfs-demo-pod ContainerName:web-server Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
    Mar 24 10:11:15.909: INFO: >>> kubeConfig: /root/.kube/config
    Mar 24 10:11:15.909: INFO: ExecWithOptions: Clientset creation
    Mar 24 10:11:15.909: INFO: ExecWithOptions: execute(POST https://192.168.49.2:8443/api/v1/namespaces/cephfs-2540/pods/csi-cephfs-demo-pod/exec?command=%2Fbin%2Fsh&command=-c&command=dd+if%3D%2Fdev%2Fzero+of%3D%2Fvar%2Flib%2Fwww%2Ftest+bs%3D1M+count%3D10+status%3Dnone&container=web-server&container=web-server&stderr=true&stdout=true)
    Mar 24 10:11:16.004: INFO: >>> kubeConfig: /root/.kube/config
    Mar 24 10:11:16.012: INFO: snapshot with name cephfs-25400 created in cephfs-2540 namespace
    Mar 24 10:11:16.012: INFO: waiting for &{{VolumeSnapshot snapshot.storage.k8s.io/v1} {cephfs-25400  cephfs-2540    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []} {{0xc000c62f20 <nil>} 0xc000ae09d0} <nil>} to be in ready state
    Mar 24 10:11:16.012: INFO: waiting for snapshot cephfs-25400 (0 seconds elapsed)
    Mar 24 10:11:18.015: INFO: waiting for snapshot cephfs-25400 (2 seconds elapsed)
    Mar 24 10:11:18.017: INFO: >>> kubeConfig: /root/.kube/config
    Mar 24 10:11:18.021: INFO: snapshot with name cephfs-25401 created in cephfs-2540 namespace
    Mar 24 10:11:18.021: INFO: waiting for &{{VolumeSnapshot snapshot.storage.k8s.io/v1} {cephfs-25401  cephfs-2540    0 0001-01-01 00:00:00 +0000 UTC <nil> <nil> map[] map[] [] [] []} {{0xc000c62f20 <nil>} 0xc000ae09d0} <nil>} to be in ready state
    Mar 24 10:11:18.021: INFO: waiting for snapshot cephfs-25401 (0 seconds elapsed)
    Mar 24 10:11:20.023: INFO: waiting for snapshot cephfs-25401 (2 seconds elapsed)
    Mar 24 10:11:20.029: INFO: ExecWithOptions {Command:[/bin/sh -c ceph fs subvolume snapshot ls myfs csi-vol-37166745-cbb9-437a-904c-7c323c29be7b --group_name=e2e --format=json] Namespace:rook-ceph PodName:rook-ceph-tools-7c4b8bb9b5-ghxvd ContainerName:rook-ceph-tools Stdin:<nil> CaptureStdout:true CaptureStderr:true PreserveWhitespace:true Quiet:false}
    Mar 24 10:11:20.029: INFO: >>> kubeConfig: /root/.kube/config
    Mar 24 10:11:20.029: INFO: ExecWithOptions: Clientset creation
    Mar 24 10:11:20.029: INFO: ExecWithOptions: execute(POST https://192.168.49.2:8443/api/v1/namespaces/rook-ceph/pods/rook-ceph-tools-7c4b8bb9b5-ghxvd/exec?command=%2Fbin%2Fsh&command=-c&command=ceph+fs+subvolume+snapshot+ls+myfs+csi-vol-37166745-cbb9-437a-904c-7c323c29be7b+--group_name%3De2e+--format%3Djson&container=rook-ceph-tools&container=rook-ceph-tools&stderr=true&stdout=true)
    Mar 24 10:11:20.597: FAIL: snapshots [[{csi-snap-1bc27d4f-c198-4723-9f0d-ec72ba0e6cb3} {csi-snap-613693ea-092a-46a9-8226-1c6e45d78bb2}]]. snapshots count 2 not matching expected count 1
```

There should really only be a single snapshot when `backingSnapshot` is enabled...